parser/parser_utils.go

```diff
@@ -193,3 +194,44 @@ func parseIdentifiers(l *lexer, t token) (err error) {
 func isDMLTerminator(t token) bool {
 	return t == tkEOF || t == tkEOS || t == tkInsert || t == tkUpdate || t == tkDelete || t == tkApply
 }
+
+// PatchQueryConsistency modifies the consistency level of a QUERY message in-place
+// by locating the consistency field directly in the frame body.
+//
+// Layout based on the CQL native protocol v4 spec:
+// /* <query: long string><consistency: short><flags: byte>... */
+func PatchQueryConsistency(body []byte, newConsistency primitive.ConsistencyLevel) error {
+	if len(body) < 6 {
+		return fmt.Errorf("body too short for QUERY")
+	}
+	queryLen := binary.BigEndian.Uint32(body[0:4])
+	offset := 4 + int(queryLen)
+	if len(body) < offset+2 {
+		return fmt.Errorf("not enough bytes to patch QUERY consistency")
+	}
+	binary.BigEndian.PutUint16(body[offset:offset+2], uint16(newConsistency))
+	return nil
+}
+
+// PatchExecuteConsistency modifies the consistency level of an EXECUTE message in-place
+// by locating the consistency field directly after the prepared statement ID.
+//
+// Layout based on the CQL native protocol v4 spec:
+// /* <id: short bytes><consistency: short><flags: byte>... */
+func PatchExecuteConsistency(body []byte, newConsistency primitive.ConsistencyLevel) error {
+	if len(body) < 2 {
+		return fmt.Errorf("body too short for EXECUTE")
+	}
+	idLen := int(binary.BigEndian.Uint16(body[0:2]))
+	offset := 2 + idLen
+	if len(body) < offset+2 {
+		return fmt.Errorf("not enough bytes to patch EXECUTE consistency")
+	}
+	binary.BigEndian.PutUint16(body[offset:offset+2], uint16(newConsistency))
+	return nil
+}
```

Comment:
Semi-nit, likely for future work: I'm wondering if we can pull the message-specific stuff out into logic which selects an offset for each specific message type.  We then have a single PatchConsistency() function (or something similar) which takes those offsets and actually makes the change.

Only argument for it is that it (perhaps) makes it clearer in code what's actually going on here.  But again, this shouldn't hold up getting this PR out... I'm mentioning this more as a note to future implementors to consider.